cmd/agent/app/integrations.go

```diff
@@ -25,11 +25,21 @@ import (
 )
 
 const (
-	constraintsFile = "agent_requirements.txt"
-	tufConfigFile   = "public-tuf-config.json"
-	tufPkgPattern   = "datadog-.*"
+	constraintsFile       = "agent_requirements.txt"
+	tufConfigFile         = "public-tuf-config.json"
+	tufPkgPattern         = "datadog-.*"
+	pipCheckOutputPattern = "%s \\d+\\.\\d+\\.\\d+ has requirement " +
+		"datadog-checks-base..?\\d+\\.\\d+\\.\\d+, " +
+		"but you have datadog-checks-base \\d+\\.\\d+\\.\\d+"
```

Comment:
you're probably aware of this already, but I'm concerned about the maintainability of using a complex regexp on a human-readable message that's output by a pip command: this will likely change in future pip versions and break this silently 😕 

Instead of having complex logic around `pip check`, could we run it twice (once before making changes, once after) without parsing the output, and fail on the first run if `pip check` finds anything wrong in the environment?

Reply:
i like this idea